### PR TITLE
Add bridge stopped and resumed events supports

### LINKED ISSUE #5785: Core to listen to the Bridge_Stop and Bridge_Start events
The core should listen to the Bridge_Stop and Bridge_Start events.

Once a bridge stop is received, the network should stop issuing any withdrawal.

Issuance of withdrawals should restart once the bridge_Start event is received.

### PR DESCRIPTION
**Pull request by jeremyletang**
Add support for bridge stopped and resumed events.
Stop issuing withdrawals if the bridge is stopped.
Also add BridgeView type in bridge package (start removal of the ethclient in assets).

erc20 asset don't need and eth client or signer anymore. testing of the banking package should be eased from now on.

Lots of changes in the datanode side of things as well, as I could not re-generate the mocks due to some interface being broken because mockgen not supporting generics.

close #5785

### CODE REVIEW

File: core/types/snapshot_nodes.go [hunk 1]
```diff
@@ -2025,12 +2079,14 @@ func BankingAssetActionsFromProto(aa *snapshot.BankingAssetActions) *BankingAsse
...
-		BlockNumber: a.BlockNumber,
-		TxIndex:     a.TxIndex,
-		Hash:        a.Hash,
+		ID:            a.Id,
+		State:         a.State,
+		Asset:         a.Asset,
+		BlockNumber:   a.BlockNumber,
+		TxIndex:       a.TxIndex,
+		Hash:          a.Hash,
+		BridgeStopped: a.Erc20BridgeResumed,
```
Comment (ze97286) [inline core/types/snapshot_nodes.go:2088]: nah

Comment (jeremyletang) [reply core/types/snapshot_nodes.go]: yeah

File: core/banking/bridge_stop_resume.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+) {
+	if b.isNewerEvent(block, logIndex) {
+		b.active, b.block, b.logIndex = true, block, logIndex
+	}
+}
+
+func (b *bridgeState) isNewerEvent(
+	block, logIndex uint64,
+) bool {
+	if block > b.block {
```
Comment (ze97286) [inline core/banking/bridge_stop_resume.go:83]:
```
if block == b.block {
    return logIndex > b.logIndex 
}
return block > b.block

```

Comment (jeremyletang) [reply core/banking/bridge_stop_resume.go:83]: fair enough

File: core/banking/checkpoint.go [hunk 3]
```diff
@@ -53,13 +54,32 @@ func (e *Engine) Load(ctx context.Context, data []byte) error {
...
 
+func (e *Engine) loadBridgeState(state *checkpoint.BridgeState) {
+	// this would eventually be nil if we restore from a checkpoint
+	// which have been produce from an old version of the core.
+	// we set it to active by default in the case
+	if state == nil {
+		e.bridgeState = &bridgeState{
+			active: true,
+		}
+	} else {
```
Comment (EVODelavega) [inline core/banking/checkpoint.go:74]: nit-pick: `return` to remove `else`

File: core/banking/snapshot.go [hunk 4]
```diff
@@ -280,6 +321,22 @@ func (e *Engine) restoreScheduledTransfers(ctx context.Context, transfers []*che
 	return err
 }
 
+func (e *Engine) restoreBridgeState(ctx context.Context, state *types.BankingBridgeState, p *types.Payload) error {
+	var err error
```
Comment (EVODelavega) [inline core/banking/snapshot.go:325]: this `err` never seems to get set, why is it here?

**Review by EVODelavega — DISMISSED** [review]

File: core/banking/snapshot.go (hunk 4 above)
Comment (jeremyletang) [reply core/banking/snapshot.go:325]: it is but at the end of the function. cleaning up just to make it more clear

File: core/banking/checkpoint.go (hunk 3 above)
Comment (jeremyletang) [reply core/banking/checkpoint.go]: addressed

File: core/banking/snapshot.go (hunk 4 above)
Comment (jeremyletang) [reply core/banking/snapshot.go]: addressed

**Review by ze97286 — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**